File: src/main/java/com/procurement/submission/model/ocds/BidStatus.java
package com.procurement.submission.model.ocds;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.HashMap;
import java.util.Map;

public enum BidStatus {
    INVITED("invited"),
    PENDING("pending"),
    VALID("valid"),
    DISQUALIFIED("disqualified"),
    WITHDRAWN("withdrawn");

    private static final Map<String, BidStatus> CONSTANTS = new HashMap<String, BidStatus>();
    private final String value;

    static {
        for (final BidStatus c : values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    BidStatus(final String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }

    @JsonCreator
    public static BidStatus fromValue(final String value) {
        final BidStatus constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new IllegalArgumentException(value);
        }
        return constant;
    }
}

File: src/main/kotlin/com/procurement/submission/domain/model/enums/BidStatus.kt
package com.procurement.submission.domain.model.enums

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class BidStatus(@JsonValue override val key: String) : EnumElementProvider.Key {
    INVITED("invited"),
    PENDING("pending"),
    VALID("valid"),
    DISQUALIFIED("disqualified"),
    WITHDRAWN("withdrawn"),
    EMPTY("empty");

    override fun toString(): String = key

    companion object : EnumElementProvider<BidStatus>(info = info()) {

        @JvmStatic
        @JsonCreator
        fun creator(name: String) = BidStatus.orThrow(name)
    }
}